pkg/bytecode/vm_test.go

```diff
@@ -135,6 +135,66 @@ func TestBoolExpressions(t *testing.T) {
 	}
 }
 
+func TestUnaryExpression(t *testing.T) {
+	tests := []testCase{
+		{
+			name: "bang operator",
+			input: `
+			x := !true
+			x = x
+			`,
+			expectedStackTop:  false,
+			expectedConstants: []any{},
+			expectedInstructions: []Instructions{
+				mustMake(t, OpTrue),
+				mustMake(t, OpBang),
+				mustMake(t, OpSetGlobal, 0),
+				mustMake(t, OpGetGlobal, 0),
+				mustMake(t, OpSetGlobal, 0),
+			},
+		},
+		{
+			name: "minus operator",
+			input: `
+			x := -1
+			x = x
+			`,
+			expectedStackTop:  -1,
+			expectedConstants: []any{1},
+			expectedInstructions: []Instructions{
+				mustMake(t, OpConstant, 0),
+				mustMake(t, OpMinus),
+				mustMake(t, OpSetGlobal, 0),
+				mustMake(t, OpGetGlobal, 0),
+				mustMake(t, OpSetGlobal, 0),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			program, err := parser.Parse(tt.input, parser.Builtins{})
+			assert.NoError(t, err, "parser error")
+			comp := NewCompiler()
+			err = comp.Compile(program)
+			assert.NoError(t, err, "compiler error")
+			vm := NewVM(comp.Bytecode())
+			err = vm.Run()
+			assert.NoError(t, err, "runtime error")
+			stackElem := vm.lastPoppedStackElem()
+			switch expected := tt.expectedStackTop.(type) {
```

Comment:
I think at this point I'd split this up into `TestBoolUnary` and `TestNumUnary` top level tests.

Reply:
Taking another look at this, I don't think it makes sense to highlight unary expression tests in their own methods. So I have just included them in the relevant bool or num tests